LoopTests/LoopDataManagerTests.swift

```diff
@@ -0,0 +1,300 @@
+@testable import LoopCore
+@testable import Loop
+
+public typealias JSONDictionary = [String: Any]
+
+enum DataManagerTestType {
+    case flatAndStable
+    case highAndStable
+    case highAndRisingWithCOB
+    case lowAndFallingWithCOB
+    case lowWithLowTreatment
+    case highAndFalling
+}
+
+extension TimeZone {
+    static var fixtureTimeZone: TimeZone {
+        return TimeZone(secondsFromGMT: 25200)!
+    }
+    
+    static var utcTimeZone: TimeZone {
+        return TimeZone(secondsFromGMT: 0)!
+    }
+}
+
+extension ISO8601DateFormatter {
+    static func localTimeDate(timeZone: TimeZone = .fixtureTimeZone) -> Self {
+        let formatter = self.init()
+
+        formatter.formatOptions = .withInternetDateTime
+        formatter.formatOptions.subtract(.withTimeZone)
+        formatter.timeZone = timeZone
+
+        return formatter
+    }
+}
+
+class LoopDataManagerDosingTests: XCTestCase {
+    // MARK: Constants for testing
+    let retrospectiveCorrectionEffectDuration = TimeInterval(hours: 1)
+    let retrospectiveCorrectionGroupingInterval = 1.01
+    let retrospectiveCorrectionGroupingIntervalMultiplier = 1.01
+    let inputDataRecencyInterval = TimeInterval(minutes: 15)
+    let dateFormatter = ISO8601DateFormatter.localTimeDate()
+    
+    // MARK: Settings
+    let maxBasalRate = 5.0
+    let maxBolus = 10.0
+    
+    var suspendThreshold: GlucoseThreshold {
+        return GlucoseThreshold(unit: HKUnit.milligramsPerDeciliter, value: 75)
+    }
+    
+    var adultExponentialInsulinModel: InsulinModel = ExponentialInsulinModel(actionDuration: 21600.0, peakActivityTime: 4500.0)
+
+    var glucoseTargetRangeSchedule: GlucoseRangeSchedule {
+        return GlucoseRangeSchedule(unit: HKUnit.milligramsPerDeciliter, dailyItems: [
+            RepeatingScheduleValue(startTime: TimeInterval(0), value: DoubleRange(minValue: 100, maxValue: 110)),
+            RepeatingScheduleValue(startTime: TimeInterval(28800), value: DoubleRange(minValue: 90, maxValue: 100)),
+            RepeatingScheduleValue(startTime: TimeInterval(75600), value: DoubleRange(minValue: 100, maxValue: 110))
+        ], timeZone: .utcTimeZone)!
+    }
+    
+    // MARK: Mock stores
+    var loopDataManager: LoopDataManager!
+    
+    func setUp(for test: DataManagerTestType) {
+        let settings = LoopSettings(
+            dosingEnabled: false,
+            glucoseTargetRangeSchedule: glucoseTargetRangeSchedule,
+            maximumBasalRatePerHour: maxBasalRate,
+            maximumBolus: maxBolus,
+            suspendThreshold: suspendThreshold
+        )
+        
+        let doseStore = MockDoseStore(for: test)
+        doseStore.basalProfileApplyingOverrideHistory = loadBasalRateScheduleFixture("basal_profile")
+        let glucoseStore = MockGlucoseStore(for: test)
+        let carbStore = MockCarbStore(for: test)
+        
+        let currentDate = glucoseStore.latestGlucose!.startDate
+        
+        loopDataManager = LoopDataManager(
```

Comment:
Initialize the LDM with mock stores that return different effects depending on which `DataManagerTestType` they're set up with